### ISSUE TITLE
Memory leak in `SystemProperties.sChangeCallbacks` & `AccessibilityManagerService.mContext`

### ISSUE THREAD

**Issue by raviagarwal7** [robolectric/robolectric#3077]
### Description
Both leaks the application

static field `SystemProperties.sChangeCallbacks` is not getting reset after test runs.
Resetting it to a empty list in the base class fixes the issue.
```
    @After
    public void tearDownGlobal() {
        // Reset System Properties change callbacks
        ReflectionHelpers.setStaticField(SystemProperties.class, "sChangeCallbacks", new ArrayList());
    }
```

Same is for `AccessibilityManagerService.mContext` but I think it could be related to https://github.com/robolectric/robolectric/issues/2791
<img width="815" alt="screen shot 2017-05-10 at 5 54 36 pm" src="https://cloud.githubusercontent.com/assets/1913956/25927504/3f52bada-35ab-11e7-9a9e-50ccd80329cd.png">

### Robolectric & Android Version
Robolectric version 3.3.2
Android Version 21
Using buck runner for running tests

**Comment by raviagarwal7** [comment]
cc @xian @kageiit

**Comment by jongerrish** [comment]
Thanks Ravi,

Would you be up for submitting a PR to fix this? I'll happily merge it for
you.

**Comment by raviagarwal7** [comment]
sure, will create one.

**Comment by kageiit** [comment]
@xian Seems like this should be fixed in 8f3490a5b77424482f82f97867090e83cf46e863 ?

If so, can we close this? Thanks!

**Comment by utzcoz** [comment]
Fixed by https://github.com/robolectric/robolectric/commit/8f3490a5b77424482f82f97867090e83cf46e863. Close it.
